src/SDKs/Azure.Base/data-plane/Azure.Base/Pipeline/ResponseClassifier.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+
+namespace Azure.Base.Pipeline
+{
+    public abstract class ResponseClassifier
```

Comment:
This adds a new concept/type, which you know I have an aversion to. But it's possible that it's absolutely needed. i.e. I am on the fence.

Reply:
It removes a need to flow the same values/delegates to multiple components of the system.